students/charlie_england/session05/mailroom_part3.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+#refactor to have the donors dict be 2 lists
+import sys
+# Donors dict Name:[[list of donations],[sum donations, num donations, average donations]]
+donors = {"Chris Christly": [[1000.21, 250.80],[]], "Bob Barley": [[800.33],[]], "Nick Nilly": [[
+    500000.12, 250000.55, 750000],[]], "Julia July": [[200.80],[]], "Jose Hooray": [[500000, 1000000, 750000],[]]}
+
+
+def main_menu():
+    '''
+    Main menu for the program.
+    Will loop through and requests input until 3 is entered and then quits.
+    1 will allow an addition of a donator, a new donation and the printing of a thank you letter
+    2 will print a report of the donators and their donations as well as their average donation.
+    3 will send letters to all donors by making .txt files for each
+    4 will quit the program
+    '''
+    switcher = {
+        "1": thank_you,
+        "2": report,
+        "3": make_letters,
+        "4": halt
+    }
+    while True:
+        try:
+            argument = input("""What would you like to do?
+            1: Send a Thank You to a single donor
+            2: Create a Report
+            3: Send letters to all donors
+            4: Quit
+            >>>""").strip()
+            func = switcher.get(argument, "Invalid Result")
+            print(func())
+        except TypeError:
+            print(f"You entered {argument}, please enter 1,2,3 or 4.")
+
+
+def halt():
+    print("Quitting...")
+    sys.exit()
+
+def thank_you():
+    """
+    will ask for a name input or a list.
+    Will print a list of names if list is entered
+    If name is in the dict, will ask for a donation amount, if name is not in dict, will ask if you would like to add a new name and will ask for a donation amount
+    """
+    while True:
+        name_input = input(
+            "Please enter a full name, 'list' for list of names, or back to go back to previous menu \n>>>").strip()
+        if name_input in donors.keys():
+            # Ask for donation amount and append to key in dict
+            donors[name_input][0].append(
+                float(input("Please enter a donation amount: >>> ")))
```

Comment:
Try to add some error handling here in the event that the user does not submit a number